src/tester.rs

```diff
@@ -658,8 +664,26 @@ fn run_cmd(
         .spawn()
         .unwrap_or_else(|_| fatal(&format!("Couldn't run command {:?}.", cmd)));
 
-    let mut stderr = FileDescriptor::dup(child.stderr.as_ref().unwrap()).unwrap();
     let mut stdout = FileDescriptor::dup(child.stdout.as_ref().unwrap()).unwrap();
+    let mut stderr = FileDescriptor::dup(child.stderr.as_ref().unwrap()).unwrap();
+
+    let stdout_fd = stdout.as_socket_descriptor();
+    let stderr_fd = stderr.as_socket_descriptor();
+
+    let flags =
```

Comment:
Because there's so much `unwrap_or_else` in here, it might be worth breaking this out into a function that has a return type along the lines of `Result<c_int, ()>` then you'll only need one `unwrap_or_else`.

Reply:
Agreed.